### ISSUE TITLE
has_many not returning related records

### ISSUE THREAD

**Issue by jpmateo022** [MasoniteFramework/orm#360]
It seems there is a bug in @has_many. it's not returning the related records

my code

```
    @has_many('ud_job_id','ud_job_id')
    def workflows(self):
        return SoftwareUDRunWorkflows
```

**Comment by josephmancuso** [comment]
your keys don't look right to me. what does the pivot table look like?

**Comment by josephmancuso** [comment]
Can I see the records in the related tables? will need to reproduce this

**Comment by jpmateo022** [comment]
I have a two table

Jobs Table has ud_job_id field which is unique while in Workflows table ud_job_id field is a foreign table that can be referenced to multiple records.

This is actually working when Im using orator before.

**Comment by josephmancuso** [comment]
Could you share the query that is running?

**Comment by jpmateo022** [comment]
This is the query Im currently using

```

        query   = SoftwareUDRunJobs.where('ud_job_id',ud_job_id)
        query   = query.with_('device','run','workflows','schedule')
        result  = query.first()
```

The device, run, and schedule are using belongs_to and its working perfectly but when I check the workflows attribute its is an empty value. Only the workflows uses the has_many

**Comment by josephmancuso** [comment]
Ok thanks will try to replicate 🤔

**Comment by jpmateo022** [comment]
Hi @josephmancuso, Were you able to replicate this?

**Comment by josephmancuso** [comment]
No. My has many relationship was returning records

**Comment by josephmancuso** [comment]
anymore info you can supply here?

**Comment by jpmateo022** [comment]
Sorry I need to mask my org name hehe. Anyways,

The relationship of the table is **jobs(one)** to **workflows(many)**

This is my jobs table

![image](https://user-images.githubusercontent.com/26633918/106701049-faf0be00-6620-11eb-8042-e11c157939ad.png)

![image](https://user-images.githubusercontent.com/26633918/106700916-c1b84e00-6620-11eb-83e3-853c10b7f252.png)




This is the workflows table containing multiple records referencing to jobs table

![image](https://user-images.githubusercontent.com/26633918/106701114-1956b980-6621-11eb-88d5-eeb1c2a6a744.png)
![image](https://user-images.githubusercontent.com/26633918/106701254-591da100-6621-11eb-8c00-12db7d99e804.png)




and this in my code

```
from masoniteorm.relationships import belongs_to, has_many
```

![image](https://user-images.githubusercontent.com/26633918/106701580-dc3ef700-6621-11eb-85fa-ad2a8a71b466.png)
![image](https://user-images.githubusercontent.com/26633918/106700680-46569c80-6620-11eb-9d68-ceb79d72fdec.png)


So when I execute the code I got this result in postman.



![image](https://user-images.githubusercontent.com/26633918/106701435-9bdf7900-6621-11eb-8234-de8e2ff56afa.png)

**Comment by josephmancuso** [comment]
Perfect. With this info i was able to reproduce it. Will be pushing a fix shortly

**Comment by josephmancuso** [comment]
The issue actually was that the has many was not working properly when the foreign key was not the primary key of the model .. not sure how i missed that

**Comment by josephmancuso** [comment]
This is fixed in V1.0.20

**Comment by jpmateo022** [comment]
Thank you very much for the fast response.
